File: kirun-java/src/test/java/com/fincity/nocode/kirun/engine/repository/KIRunFunctionRepositoryTest.java
package com.fincity.nocode.kirun.engine.repository;

import org.junit.jupiter.api.Test;

import com.fincity.nocode.kirun.engine.function.reactive.ReactiveFunction;
import com.fincity.nocode.kirun.engine.function.system.math.MathFunctionRepository;
import com.fincity.nocode.kirun.engine.model.FunctionSignature;
import com.fincity.nocode.kirun.engine.namespaces.Namespaces;
import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveFunctionRepository;

import reactor.core.publisher.Mono;
import reactor.test.StepVerifier;

class KIRunReactiveFunctionRepositoryTest {

	@Test
	void test() {

		Mono<ReactiveFunction> fun = new MathFunctionRepository().find(Namespaces.MATH, "Absolute");

		StepVerifier.create(fun)
				.expectNextMatches(funs -> funs.getSignature().getName()
						.equals("Absolute"))
				.verifyComplete();

		Mono<FunctionSignature> signature = fun.map(ReactiveFunction::getSignature)
				.flatMap(funs -> new KIRunReactiveFunctionRepository().find(funs.getNamespace(), funs.getName()))
				.map(ReactiveFunction::getSignature);

		StepVerifier.create(signature).expectNextMatches(fs -> fs.getNamespace().equals(Namespaces.MATH))
				.verifyComplete();

		Mono<ReactiveFunction> function = new KIRunReactiveFunctionRepository().find(Namespaces.STRING, "ToString");

		StepVerifier.create(function).expectNextMatches(funs -> funs.getSignature().getName()
				.equals("ToString")).verifyComplete();

		function = new KIRunReactiveFunctionRepository().find(Namespaces.STRING, "IndexOfWithStartPoint");

		StepVerifier.create(function).expectNextMatches(funs -> funs.getSignature().getName()
				.equals("IndexOfWithStartPoint")).verifyComplete();

		function = new KIRunReactiveFunctionRepository().find(Namespaces.SYSTEM_ARRAY, "Compare");

		StepVerifier.create(function).expectNextMatches(funs -> funs.getSignature().getName()
				.equals("Compare")).verifyComplete();

		function = new KIRunReactiveFunctionRepository().find(Namespaces.MATH, "RandomInt");

		StepVerifier.create(function).expectNextMatches(funs -> funs.getSignature().getName()
				.equals("RandomInt")).verifyComplete();

		function = new KIRunReactiveFunctionRepository().find(Namespaces.MATH, "Exponential");

		StepVerifier.create(function).expectNextMatches(funs -> funs.getSignature().getName()
				.equals("Exponential")).verifyComplete();

		function = new KIRunReactiveFunctionRepository().find(Namespaces.SYSTEM, "If");

		StepVerifier.create(function).expectNextMatches(funs -> funs.getSignature().getName()
				.equals("If")).verifyComplete();
	}

}

File: kirun-js/__tests__/engine/repository/KIRunFunctionRepositoryTest.ts
import { KIRunFunctionRepository, Namespaces } from '../../../src';

test('KIRunFunctionRepository Test', async () => {
    const repo = new KIRunFunctionRepository();

    let fun = await repo.find(Namespaces.STRING, 'ToString');
    expect(fun).toBeTruthy();
    expect(fun?.getSignature().getName()).toBe('ToString');

    fun = await repo.find(Namespaces.STRING, 'IndexOfWithStartPoint');
    expect(fun).toBeTruthy();
    expect(fun?.getSignature().getName()).toBe('IndexOfWithStartPoint');

    fun = await repo.find(Namespaces.SYSTEM_ARRAY, 'Compare');
    expect(fun).toBeTruthy();
    expect(fun?.getSignature().getName()).toBe('Compare');

    fun = await repo.find(Namespaces.MATH, 'RandomInt');
    expect(fun).toBeTruthy();
    expect(fun?.getSignature().getName()).toBe('RandomInt');

    fun = await repo.find(Namespaces.MATH, 'Exponential');
    expect(fun).toBeTruthy();
    expect(fun?.getSignature().getName()).toBe('Exponential');

    fun = await repo.find(Namespaces.SYSTEM, 'If');
    expect(fun).toBeTruthy();
    expect(fun?.getSignature().getName()).toBe('If');
});
